docs/docs/integrations/providers/notdiamond.mdx

```diff
@@ -0,0 +1,41 @@
+# Not Diamond
+
+[Not Diamond](https://notdiamond.ai) offers an AI-powered model router that automatically determines which LLM is best suited to respond to any query,
+improving LLM output quality by combining multiple LLMs into a **meta-model** that learns when to call each LLM.
+
+Not Diamond supports cost and latency tradeoffs, customized router training, and real-time router personalization. Learn more via [the documentation](https://notdiamond.readme.io/).
+
+## Installation
+
+```shell
+pip install langchain notdiamond
+```
+
+## Configuration
+
+- API keys: You need API keys from [Not Diamond](https://app.notdiamond.ai/keys) and any LLM you want to use.
+
+## Quick Start
+
+Not Diamond offers `NotDiamondRoutedRunnable`, allowing you to include routing within your chain.
+
+```python
+from langchain_community.utilities.notdiamond import NotDiamondRoutedRunnable
+
+from notdiamond import NotDiamond, LLMConfig
+
+nd_client = NotDiamond(
```

Comment:
The implementation suggests that a custom `NotDiamond` client is optional, and a user can use `NotDiamondRoutedRunnable` directly with a list of model names as well. Is my understanding correct? If so, it would be great to have an example for the latter too since it's less friction for users.

Reply:
Great point, I can add this example.

Relatedly, what are your thoughts on documenting usage of `NotDiamondRunnable`, the function which only performs modelSelect? Is it worth keeping around for users who might only want to receive a routing rec? Or should I consolidate these `Runnable`s?

Reply:
I think separating them is good because the runnable is supposed to have a fixed set of public methods and we want to make sure `.invoke` either returns the response from the LLM recommended or the LLM itself. But I think `NotDiamondRunnable` should return the LLM runnable (i.e. the object returned from `self._configurable_model` in `NotDiamondRoutedRunnable`) because that would facilitate further chaining or invoking as needed

Reply:
I definitely think it's worth keeping around since it could be useful for users who just need the routing recommendation. We should document it better and provide examples though, I missed the usage of the `NotDiamondRunnable` on the first pass.

Reply:
> But I think NotDiamondRunnable should return the LLM runnable (i.e. the object returned from self._configurable_model in NotDiamondRoutedRunnable) because that would facilitate further chaining or invoking as needed

@r0ymanesco Do you mean that `model_select` should return the model, or `invoke`, or something else?

If someone wants to chain further, they broadly have two options:
 1. invoke the selected model, in which case they want `NotDiamondRoutedRunnable` anyway, or
 2. apply business logic against the selected model,

For these reasons I actually prefer the current approach, since it's less painful to check

```python
if result == "openai/gpt-4o":
    print(result)
elif "anthropic" in result:
    run_anthropic_fn(result)
```

than

```python
if isinstance(result, ChatOpenAI) and result.model_provider == "perplexity":
    ...
```

Note the latter is actually necessary - some providers actually reuse the `ChatOpenAI` client, like Perplexity.

Reply:
Yea I was referring to the option that returns the model but I see your point about the difficulty in applying logic after that. Let's go with your approach